extension/spark-doris-connector/src/main/java/org/apache/doris/spark/rest/RestService.java

```diff
@@ -0,0 +1,462 @@
+        Map<String, List<Long>> be2Tablets = selectBeForTablet(queryPlan, logger);
+        return tabletsMapToPartition(
+                cfg,
+                be2Tablets,
+                queryPlan.getOpaqued_query_plan(),
+                tableIdentifiers[0],
+                tableIdentifiers[1],
+                logger);
+    }
+
+    /**
+     * translate Doris FE response string to inner {@link QueryPlan} struct.
+     * @param response Doris FE response string
+     * @param logger {@link Logger}
+     * @return inner {@link QueryPlan} struct
+     * @throws DorisException throw when translate failed.
+     */
+    @VisibleForTesting
+    static QueryPlan getQueryPlan(String response, Logger logger) throws DorisException {
+        logger.trace("Parsing fe response to query plan.");
+        ObjectMapper mapper = new ObjectMapper();
+        QueryPlan queryPlan;
+        try {
+            queryPlan = mapper.readValue(response, QueryPlan.class);
+        } catch (JsonParseException e) {
+            String errMsg = "Doris FE's response is not a json. res: " + response;
+            logger.error(errMsg, e);
+            throw new DorisException(errMsg, e);
+        } catch (JsonMappingException e) {
+            String errMsg = "Doris FE's response cannot map to schema. res: " + response;
+            logger.error(errMsg, e);
+            throw new DorisException(errMsg, e);
+        } catch (IOException e) {
+            String errMsg = "Parse Doris FE's response to json failed. res: " + response;
+            logger.error(errMsg, e);
+            throw new DorisException(errMsg, e);
+        }
+
+        if (queryPlan == null) {
+            logger.error(SHOULD_NOT_HAPPEN_MESSAGE);
+            throw new ShouldNeverHappenException();
+        }
+
+        if (queryPlan.getStatus() != REST_RESPONSE_STATUS_OK) {
+            String errMsg = "Doris FE's response is not OK, status is " + queryPlan.getStatus();
+            logger.error(errMsg);
+            throw new DorisException(errMsg);
+        }
+        logger.debug("Parsing partition result is '{}'.", queryPlan);
+        return queryPlan;
+    }
+
+    /**
+     * select which Doris BE to get tablet data.
+     * @param queryPlan {@link QueryPlan} translated from Doris FE response
+     * @param logger {@link Logger}
+     * @return BE to tablets {@link Map}
+     * @throws DorisException throw when select failed.
+     */
+    @VisibleForTesting
+    static  Map<String, List<Long>> selectBeForTablet(QueryPlan queryPlan, Logger logger) throws DorisException {
+        logger.trace("Choice tablet targeting Doris BE.");
+        Map<String, List<Long>> be2Tablets = new HashMap<>();
+        for (Map.Entry<String, Tablet> part : queryPlan.getPartitions().entrySet()) {
+            logger.debug("Parse tablet info: '{}'.", part);
+            long tabletId;
+            try {
+                tabletId = Long.parseLong(part.getKey());
```

Comment:
tabletId must be numeric, catch such exception doesn't make sense

Reply:
reserve it just in case